### PR TITLE
Implement getAllCommitments Query to get paginated commitments

### PR DESCRIPTION
**Pull request by fredabisai**
What I did:

- Created PaginatedCommitmentResponse in CommitmentDto file
- Created ListCommitmentsQuery cqrs of commitment modules
- Create API to get all commitments

Can you help me a way forward?

### CODE REVIEW

File: src/modules/commitment/cqrs/query/list-commitments.query.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@nestjs/common';
+import { IPaginationQueryArgs, IQuery, paginate } from '@common/cqrs';
+import { FindCriteria } from '@core/repository.interface';
+import { ICommitment, ICommitmentRepository, QueryCommitment } from '@domain/commitment';
+import { IPaginatedList } from '@domain/core';
+
+@Injectable()
+export class ListCommitmentsQuery implements IQuery<IPaginationQueryArgs<FindCriteria<QueryCommitment>>> {
```
Comment (MichaelSlatteryDev) [inline src/modules/commitment/cqrs/query/list-commitments.query.ts:8]: A good thing to do for this is also make a unit test for this query. Good to give you experience in that part of the code base too 🙂

**Review by MichaelSlatteryDev — CHANGES_REQUESTED** [review]

File: src/modules/commitment/cqrs/query/list-commitments.query.ts (hunk 1 above)
Comment (fredabisai) [reply src/modules/commitment/cqrs/query/list-commitments.query.ts:8]: Hi Michael this is done

**Review by MichaelSlatteryDev — APPROVED** [review]

File: src/api/rest/commitments/v1/controller/commitment.controller.ts [hunk 2]
```diff
@@ -31,4 +34,15 @@ export class CommitmentController {
...
         return await this.cqrsHandler.execute(UpdateCommitmentCommand, { id, updateDto });
     }
+    @Get()
+    @ApiOperation({ summary: 'Retrieve a paginated list of Commitments' })
+    @ApiResponse({
+        status: 200,
+        description: 'Returns the found Commitments if any',
+        type: PaginatedCommitmentResponse,
+    })
+    @FirebaseGuard(UserRole.ADMIN)
```
Comment (Xazin) [inline src/api/rest/commitments/v1/controller/commitment.controller.ts:44]:
```suggestion
    @FirebaseGuard(UserRole.MODERATOR, UserRole.ADMIN)
```

Comment (fredabisai) [reply src/api/rest/commitments/v1/controller/commitment.controller.ts]: @Xazin this is done

**Pull request merged**